airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -1554,6 +1578,66 @@ def create_inline_schema_loader(
     ) -> InlineSchemaLoader:
         return InlineSchemaLoader(schema=model.schema_ or {}, parameters={})
 
+    @staticmethod
+    def create_types_pair(model: TypesPairModel, **kwargs: Any) -> TypesPair:
+        return TypesPair(target_type=model.target_type, current_type=model.current_type)
+
+    def create_schema_type_identifier(
+        self, model: SchemaTypeIdentifierModel, config: Config, **kwargs: Any
+    ) -> SchemaTypeIdentifier:
+        types_map = []
+        if model.types_map:
+            types_map.extend(
```

Comment:
It looks like we have similar behavior in [DpathExtractor](https://github.com/airbytehq/airbyte-python-cdk/blob/cd1bd1c7119a31986b85c64cd88053cd4400dd1d/airbyte_cdk/sources/declarative/extractors/dpath_extractor.py#L62-L64) and [ParentStreamConfig](https://github.com/airbytehq/airbyte-python-cdk/blob/cd1bd1c7119a31986b85c64cd88053cd4400dd1d/airbyte_cdk/sources/declarative/partition_routers/substream_partition_router.py#L57-L60).

It might be a good idea to create a helper function as a follow-up. For now, I propose leaving it as is and revisiting this after merging this PR